lib/site-inspector.rb

```diff
@@ -25,22 +25,24 @@ def self.load_data(name)
     YAML.load_file File.expand_path "./data/#{name}.yml", File.dirname(__FILE__)
   end
 
-  def initialize(domain)
+  # makes no network requests
+  def initialize(domain, options = {})
     domain = domain.downcase
     domain = domain.sub /^https?\:/, ""
     domain = domain.sub /^\/+/, ""
     domain = domain.sub /^www\./, ""
     @uri = Addressable::URI.parse "//#{domain}"
     @domain = PublicSuffix.parse @uri.host
+    @timeout = options[:timeout] || 10
   end
 
   def inspect
     "<SiteInspector domain=\"#{domain}\">"
   end
 
-  def uri(ssl=https?,www=www?)
+  def uri(ssl=enforce_https?,www=www?)
```

Comment:
This is a change to **both modes**: This means that the "canonical" URI will only end up as `https://` if the domain forces HTTPS. For example, the canonical URL for nasa.gov was `https://www.nasa.gov` -- however, I'd argue that the canonical URL is `http://www.nasa.gov`, as that's what Google brings you to and how NASA links to stuff.

So now, the canonical URL will be `https://` only if the domain enforces HTTPS. I still need to add a test for this behavior, and if you'd rather I spin it off from here, I can.